src/dbnode/encoding/encoder_pool.go

```diff
@@ -44,5 +44,6 @@ func (p *encoderPool) Get() Encoder {
 }
 
 func (p *encoderPool) Put(encoder Encoder) {
+	encoder.SetSchema(nil)
```

Comment:
Similar comment here:

This should just be handled by `Reset(...)` then you don't need to add extra logic to the pool gets/puts.

Making pools know more about their data structures is a pattern we've tried to avoid, encapsulating that logic in `Reset(...)` for each separate data structure helps make these pools simple.

Reply:
ditto